lib/lightning/version_control/project_repo_connection.ex

```diff
@@ -26,20 +25,69 @@ defmodule Lightning.VersionControl.ProjectRepoConnection do
     field :github_installation_id, :string
     field :repo, :string
     field :branch, :string
+    field :access_token, :binary
+    field :config_path, :string
+    field :accept, :boolean, virtual: true
+    field :sync_direction, Ecto.Enum, values: [:deploy, :pull], virtual: true
     belongs_to :project, Project
-    belongs_to :user, User
 
     timestamps()
   end
 
-  @fields ~w(github_installation_id repo branch)a
-  @required_fields ~w(user_id  project_id)a
+  @required_fields ~w(github_installation_id repo branch project_id)a
+  @other_fields ~w(config_path)a
+
   def changeset(project_repo_connection, attrs) do
     project_repo_connection
-    |> cast(attrs, @fields ++ @required_fields)
+    |> cast(attrs, @required_fields ++ @other_fields)
     |> validate_required(@required_fields)
     |> unique_constraint(:project_id,
       message: "project already has a repo connection"
     )
   end
+
+  def configure_changeset(project_repo_connection, attrs) do
+    project_repo_connection
+    |> changeset(attrs)
+    |> cast(attrs, [:sync_direction, :accept])
+    |> validate_required([:sync_direction, :accept])
+    |> validate_change(:accept, fn :accept, accept ->
+      if accept do
+        []
+      else
+        [accept: "please tick the box"]
+      end
+    end)
+  end
+
+  def reconfigure_changeset(project_repo_connection, attrs) do
+    project_repo_connection
+    |> cast(attrs, [:sync_direction, :accept])
+    |> validate_required([:sync_direction, :accept])
+  end
+
+  def create_changeset(project_repo_connection, attrs) do
+    changeset = configure_changeset(project_repo_connection, attrs)
+
+    if changeset.valid? do
+      project_id = get_field(changeset, :project_id)
+
+      token = "prc_" <> generate_access_token(project_id)
+
+      put_change(changeset, :access_token, token)
+    else
+      changeset
+    end
+  end
+
+  defp generate_access_token(project_id) do
+    Joken.generate_and_sign!(default_claims(skip: [:exp]), %{
+      "project_id" => project_id
+    })
+  end
```

Comment:
Tell me more about this token, how would we rotate it and where is the signing secret? If it's using the default in `config.exs` we need to enforce using one from an env variable (like we do for the worker)